format/format.go

```diff
@@ -123,3 +135,76 @@ func commentFromText(orig ast.Node, text string) *ast.CommentGroup {
 	}
 	return group
 }
+
+func formatStruct(st *ast.StructType) error {
+	if st.Fields == nil {
+		return nil
+	}
+	// Find which struct fields require sequence numbers, and
+	// keep a record of which sequence numbers are already used.
+	usedSequences := []int{}
+	fieldsWithoutSequence := []*ast.Field{}
+	for _, f := range st.Fields.List {
+		tag := f.Tag
+		if tag == nil {
+			fieldsWithoutSequence = append(fieldsWithoutSequence, f)
+			continue
+		}
+		// Can skip the error here because we've already parsed the file.
+		str, _ := strconv.Unquote(tag.Value)
+		stag := reflect.StructTag(str)
+		val, ok := stag.Lookup("pb")
+		// If there isn't a 'pb' tag present.
+		if !ok {
+			fieldsWithoutSequence = append(fieldsWithoutSequence, f)
+			continue
+		}
+		// If there was a 'pb' tag, but it wasn't empty, return an error.
+		// It is a bit difficult to add in the sequence number if the 'pb'
+		// tag already exists.
+		if ok && val == "" {
+			return fmt.Errorf("struct field tag for pb was empty, please remove or add sequence number")
+		}
+		// If there isn't a number in 'pb' then return an error.
+		i, err := strconv.Atoi(val)
+		if err != nil {
+			return fmt.Errorf("struct field tag for pb contains a non-number %q", val)
```

Comment:
We don't catch this error in `gunk generate`, do we? We probably should too. Perhaps add a TODO.

Reply:
Done.